### ISSUE TITLE
yet another cmpIgnoreStyle bug

### ISSUE THREAD

**Issue by krux02** [nim-lang/Nim#7686]
```Nim
import strutils

type
  MyEnum = enum
    A,
    a

doAssert parseEnum[MyEnum]("A") == A
doAssert parseEnum[MyEnum]("a") == a
```
you guessed it, ``strutils.parseEnum`` uses ``strutils.cmpIgnoreStyle`` to check for identifier equality.

@Araq I offered you to fix ``strutils.cmpIgnoreStyle``, but you didn't want it.

**Comment by Araq** [comment]
It's not wrong for me. I use ``parseEnum`` for command line options where I really want ``cmpIgnoreStyle``. You guessed it, I'm aware of it.

**Comment by krux02** [comment]
I never said case-only distinctions in enum values are a good idea, but they are possible and that is enough reason to me to support them properly. But just because you find this wrong implementation of `parseEnum` in it's current state useful, doesn't justify to enforce this wrong implementation on everybody else.

**Comment by ringabout** [comment]
fixed by #14046
works in Nim 1.4
